src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -357,61 +357,26 @@ public UnmanagedMemoryStream ReadStream(CancellationToken cancellationToken)
             }
         }
 
-        public Task<Stream> ReadStreamAsync(CancellationToken cancellationToken = default)
-        {
-            // See commentary in ReadTextAsync for why this is implemented this way.
-            return Task.Factory.StartNew<Stream>(() => ReadStream(cancellationToken), cancellationToken, TaskCreationOptions.None, TaskScheduler.Default);
-        }
-
-        public void WriteStream(Stream stream, CancellationToken cancellationToken = default)
-        {
-            // The Wait() here will not actually block, since with useAsync: false, the
-            // entire operation will already be done when WaitStreamMaybeAsync completes.
-            WriteStreamMaybeAsync(stream, useAsync: false, cancellationToken: cancellationToken).GetAwaiter().GetResult();
-        }
-
-        public Task WriteStreamAsync(Stream stream, CancellationToken cancellationToken = default)
-            => WriteStreamMaybeAsync(stream, useAsync: true, cancellationToken: cancellationToken);
-
-        private async Task WriteStreamMaybeAsync(Stream stream, bool useAsync, CancellationToken cancellationToken)
+        public void WriteStream(Stream stream, CancellationToken cancellationToken)
         {
             if (_memoryMappedInfo != null)
-            {
                 throw new InvalidOperationException(WorkspacesResources.Temporary_storage_cannot_be_written_more_than_once);
-            }
 
             using (Logger.LogBlock(FunctionId.TemporaryStorageServiceFactory_WriteStream, cancellationToken))
             {
                 var size = stream.Length;
                 _memoryMappedInfo = _service.CreateTemporaryStorage(size);
                 using var viewStream = _memoryMappedInfo.CreateWritableStream();
 
-                var buffer = SharedPools.ByteArray.Allocate();
-                try
+                using var pooledObject = SharedPools.ByteArray.GetPooledObject();
+                var buffer = pooledObject.Object;
+                while (true)
```

Comment:
>while (true) [](http://example.com/codeflow?start=16&length=12)

nit: I know it requires duplicating a line, but loops with real conditions read better for me, eg:

var count = stream.Read(buffer, 0, buffer.Length);
while(count > 0)
{
    viewStream.Write(buffer, 0, count);
    count = stream.Read(buffer, 0, buffer.Length);
}

Reply:
Can i do in a followup?  I'm liking it when PRs are low concept.

Reply:
For stylistic requests, the answer to that is always yes from me